amy/dashboard/tests/test_get_involved_views.py

```diff
@@ -0,0 +1,234 @@
+            name="GitHub Contribution",
+            defaults={
+                "display_name": "GitHub Contribution",
+                "url_required": True,
+                "date_required": True,
+            },
+        )
+        self.involvement_to_be_archived, _ = Involvement.objects.get_or_create(
+            name="To be archived",
+            defaults={
+                "display_name": "To be archived",
+                "url_required": True,
+                "date_required": True,
+            },
+        )
+        self.involvement_other, _ = Involvement.objects.get_or_create(
+            name="Other",
+            defaults={
+                "display_name": "Other",
+                "notes_required": True,
+            },
+        )
+
+    def test_create_view_loads(self):
+        # Act
+        rv = self.client.get(reverse("getinvolved_add"))
+
+        # Assert
+        self.assertEqual(rv.status_code, 200)
+
+    def test_create_view_does_not_show_archived_involvements(self):
+        # Arrange
+        self.involvement_to_be_archived.archive()
+
+        # Act
+        rv = self.client.get(reverse("getinvolved_add"))
+
+        # Assert
+        self.assertEqual(rv.status_code, 200)
+        choices = [
+            c[0].instance.pk
+            for c in rv.context["form"].fields["involvement_type"].choices
+        ]
+        self.assertEqual(
+            choices, [self.github_contribution.pk, self.involvement_other.pk]
+        )
+
+    def test_create_view_does_not_show_admin_fields(self):
+        # Act
+        rv = self.client.get(reverse("getinvolved_add"))
+
+        # Assert
+        self.assertEqual(rv.status_code, 200)
+        expected_fields = {"involvement_type", "date", "url", "trainee_notes"}
+        self.assertSetEqual(set(rv.context["form"].fields.keys()), expected_fields)
+
+    def test_create_view_works(self):
+        # Arrange
+        data = {
+            "involvement_type": self.github_contribution.pk,
+            "url": "https://example.org",
+            "date": "2023-07-27",
+        }
+
+        # Act
+        rv = self.client.post(reverse("getinvolved_add"), data, follow=True)
+
+        # Assert
+        self.assertEqual(rv.status_code, 200)
+        self.assertEqual(rv.resolver_match.view_name, "training-progress")
+        self.assertContains(
+            rv, "Your Get Involved submission will be reviewed within 7-10 days."
+        )
+        got = list(
+            TrainingProgress.objects.values_list(
+                "state", "trainee", "url", "requirement", "involvement_type", "date"
+            )
+        )
+        expected = [
+            (
+                "n",
+                self.admin.pk,
+                "https://example.org",
+                self.get_involved.pk,
+                self.github_contribution.pk,
+                date(2023, 7, 27),
+            )
+        ]
+        self.assertEqual(got, expected)
+
+    def test_create_view_submission_invalid_notes(self):
```

Comment:
I feel like this is testing form validation and could be moved to form-specific test (which will perform faster than this).